src/test/codewhispererChat/editor/context/focusArea/focusAreaExtractor.test.ts

```diff
@@ -0,0 +1,312 @@
+        // Stub originSelection and extendedCodeBlockRange
+        const originSelection = {
+            start: {
+                line: 0,
+                character: 0,
+            },
+            end: {
+                line: 0,
+                character: 0,
+            },
+        }
+
+        const extendedCodeBlockRange = {
+            start: {
+                line: 2,
+                character: 0,
+            },
+            end: {
+                line: 3,
+                character: 0,
+            },
+        }
+
+        const focusAreaContextExtractor = new FocusAreaContextExtractor()
+        const method = Reflect.get(focusAreaContextExtractor, 'getSelectionInsideExtendedCodeBlock')
+
+        const result = method.call(focusAreaContextExtractor, originSelection, extendedCodeBlockRange)
+
+        assert.strictEqual(result, undefined)
+    })
+
+    it('returns adjusted selection if selection spans multiple lines', () => {
+        // Stub originSelection and extendedCodeBlockRange
+        const originSelection = {
+            start: {
+                line: 10,
+                character: 5,
+            },
+            end: {
+                line: 20,
+                character: 4,
+            },
+        }
+
+        const extendedCodeBlockRange = {
+            start: {
+                line: 9,
+                character: 0,
+            },
+            end: {
+                line: 25,
+                character: 0,
+            },
+        }
+
+        const focusAreaContextExtractor = new FocusAreaContextExtractor()
+        const method = Reflect.get(focusAreaContextExtractor, 'getSelectionInsideExtendedCodeBlock')
+
+        const result = method.call(focusAreaContextExtractor, originSelection, extendedCodeBlockRange)
+
+        checkIfExpectedRangeEqualActualRange(
+            {
+                start: {
+                    line: 1,
+                    character: 5,
+                },
+                end: {
+                    line: 11,
+                    character: 4,
+                },
+            },
+            result
+        )
+    })
+})
+
+describe('getExtendedCodeBlockRange', () => {
+    it('extends range to the beginning of the line if length is below limit', () => {
+        // Arrange
+        const document: TextDocument = createMockDocument('01234567\n0123456789')
+        const range = {
+            start: {
+                line: 1,
+                character: 8,
+            },
+            end: {
+                line: 1,
+                character: 9,
+            },
+        }
+        const limit = 10
+
+        // Act
+        const focusAreaContextExtractor = new FocusAreaContextExtractor()
+        const method = Reflect.get(focusAreaContextExtractor, 'getExtendedCodeBlockRange')
+        const result = method.call(focusAreaContextExtractor, document, range, limit)
```

Comment:
nit: is there a reason for this special syntax and not just
```
const focusAreaContextExtractor = new FocusAreaContextExtractor()
const result = focusAreaContextExtractor. getExtendedCodeBlockRange(document, range, limit)
```

Reply:
Yes, it's a private method, we cannot call it like this^